src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBase.cs

```diff
@@ -173,7 +173,7 @@ BindingFlags invokeAttr
                 {
                     RuntimeType argType = (RuntimeType)arg.GetType();
 
-                    if (ReferenceEquals(argType, sigType))
+                    if (ReferenceEquals(argType, sigType) && !argType.Equals(typeof(Missing)))
```

Comment:
```suggestion
                    if (ReferenceEquals(argType, sigType) && argType != typeof(Missing))
```
The JIT has a special optimization for Type operators == and != that will kick in here. Equals method does not have these optimizations.

Reply:
Actually I did use that at first but it did not work, somehow it evaluated to false when Missing was `argType`, the operator is not overloaded in `RuntimeType` not sure which implementation called

Reply:
Comparing the arg instance directly against `Type.Missing` should be just as fast and is what the original code did and my recent commit.